### PR TITLE
refactor(llm_provider): remove LLM_PROVIDER from config contract

### LINKED ISSUE #118: test: cover agent-runtime, agent-tools, and Python skill modules
### Background / 背景

Follows up on #42. A coverage audit of the current codebase shows that several critical subsystems have **zero dedicated tests** despite containing core business logic. This issue tracks the highest-priority gaps.

跟进 #42。对当前代码库的覆盖率审计表明，若干关键子系统虽然包含核心业务逻辑，却完全没有专项测试。本 issue 追踪优先级最高的覆盖盲区。

### Current State / 现状

| Module | Source Files | Test Files | Status |
|--------|:---:|:---:|--------|
| `backend/src/agent-runtime/` | 19 TS | 0 | No coverage |
| `backend/src/agent-tools/` | 9 TS | 0 | No coverage |
| `backend/src/agent-skills/analysis/` | ~21 Py | 0 | No coverage |
| `backend/src/agent-skills/code-check/` | ~7 Py | 0 | No coverage |
| `backend/src/agent-skills/data-input/converters/` | ~6 Py | 0 | No coverage |
| `backend/src/services/` (untested subset) | ~18 TS | 0 | No coverage |
| `backend/src/api/` (untested routes) | 6 TS | 0 | No coverage |
| `backend/src/agent-skills/material/` | 2 Py | 0 | No coverage |

### Tasks / 任务清单

**Phase 1 — Agent runtime and tools (TypeScript)**
- [ ] Add unit tests for `backend/src/agent-runtime/` (registry, loader, executor, manifest-schema, tool-registry)
- [ ] Add unit tests for `backend/src/agent-tools/builtin/` (draft-model, validate-model, run-analysis, run-code-check, convert-model, generate-report, update-model)

**Phase 2 — Python analysis and code-check modules**
- [ ] Add unit tests for `backend/src/agent-skills/analysis/` (OpenSees static/dynamic/seismic/nonlinear adapters, simplified engines, runtime)
- [ ] Add unit tests for `backend/src/agent-skills/code-check/` (GB50009, GB50010, GB50011, GB50017, JGJ3 runners)
- [ ] Add unit tests for `backend/src/agent-skills/data-input/converters/` (model conversion pipelines)
- [ ] Add unit tests for `backend/src/agent-skills/material/` (concrete, steel material definitions)

**Phase 3 — Services and API routes (TypeScript)**
- [ ] Add unit tests for untested `backend/src/services/` files (agent-capability, agent-context, agent-session, agent-router, agent-policy, agent-validation, code-check-execution, structure-protocol-execution)
- [ ] Add integration tests for untested `backend/src/api/` routes (agent, analysis, analysis-engine, chat)

### Acceptance Criteria / 验收标准

- Every module listed above has at least one test file covering its main entry point and error paths
- New tests follow existing conventions (Jest `.test.mjs` for backend TS, `unittest` for Python)
- All new tests pass in CI without flakiness
- No regression in existing test suites

- 上述每个模块至少有一个测试文件，覆盖主入口和错误路径
- 新测试遵循现有约定（后端 TS 用 Jest `.test.mjs`，Python 用 `unittest`）
- 所有新测试在 CI 中稳定通过
- 不引入已有测试套件的回归

### PR DESCRIPTION
**Pull request by qinsz01**
## Summary

Describe the problem and fix in 2–5 bullets:

- Problem: the repository still exposed `LLM_PROVIDER` even though runtime LLM calls already went through a single OpenAI-compatible client path, and LLM integration cases could still fail on one-off semantic drift because the runner only retried transient upstream errors.
- Why it matters: stale provider branching made config, docs, tests, and CI drift away from real runtime behavior, while flaky LLM case failures created noisy Actions signal.
- What changed: removed `LLM_PROVIDER` / `--llm-provider` from backend config, chat logging context, CLI `docker-install`, Docker Compose, CI workflows, regression/integration/frontend test envs, and bilingual docs; added focused backend/env contract coverage; updated the LLM integration runner so every case gets up to 3 total attempts.
- What did NOT change (scope boundary): default `LLM_MODEL` and `LLM_BASE_URL` values, the underlying LangChain OpenAI client path, and skill-local runtime behavior outside the test harness.

## Change Type (select all)

- [x] Bug fix
- [ ] Feature
- [x] Refactor required for the fix
- [x] Docs
- [ ] Security hardening
- [x] Chore/infra

## Scope (select all touched areas)

- [x] Gateway / orchestration
- [ ] Skills / tool execution
- [ ] Auth / tokens
- [ ] Memory / storage
- [x] Integrations
- [x] API / contracts
- [x] UI / DX
- [x] CI/CD / infra

## Linked Issue/PR

- Closes # N/A
- Related # N/A
- [x] This PR fixes a bug or regression

## Root Cause / Regression History (if applicable)

For bug fixes or regressions, explain why this happened, not just what changed. Otherwise write `N/A`. If the cause is unclear, write `Unknown`.

- Root cause: the config layer preserved an older provider-selection abstraction after runtime invocation had already converged on a single OpenAI-compatible `ChatOpenAI` path, and the LLM integration harness only retried transient transport failures instead of case-level semantic drift.
- Missing detection / guardrail: there was no explicit regression test asserting that backend config should not expose `llmProvider`, no contract test enforcing the reduced CLI/bootstrap surface, and no focused retry test covering deterministic case failures in the LLM runner.
- Prior context (`git blame`, prior PR, issue, or refactor if known): `upstream/master` before this branch still included provider-specific defaults and CLI prompts in `backend/src/config/index.ts` and `scripts/cli/main.js`; `tests/llm-integration/runner.cjs` wrapped cases with a retry helper that only retried transient upstream errors.
- Why this regressed now: the codebase gradually simplified around a single transport, but the env contract and docs were not collapsed with it; during PR validation, a flaky LLM extraction case passed on rerun and exposed that the runner-level retry policy still did not match real model nondeterminism.
- If unknown, what was ruled out: N/A.

## Regression Test Plan (if applicable)

For bug fixes or regressions, name the smallest reliable test coverage that should have caught this. Otherwise write `N/A`.

- Coverage level that should have caught this:
  - [x] Unit test
  - [x] Seam / integration test
  - [ ] End-to-end test
  - [ ] Existing coverage already sufficient
- Target test or file: `backend/tests/config.llm-env.test.mjs`, `backend/tests/env-compose.contract.test.mjs`, and `tests/llm-integration/lib/retry.test.cjs`.
- Scenario the test should lock in: backend config should expose only direct OpenAI-compatible env defaults, `docker-install` help/contract should no longer mention `--llm-provider`, and the LLM integration runner should allow up to 3 total attempts for a case when case-level retries are enabled.
- Why this is the smallest reliable guardrail: one backend config assertion, one CLI/env contract assertion, and one focused retry unit test cover the reduced config surface plus the new runner behavior without requiring a full external LLM run.
- Existing test that already covers this (if any): `node tests/runner.mjs backend-regression` now exercises the updated env contract transitively after the focused assertions pass.
- If no new test is added, why not: N/A.

## User-visible / Behavior Changes

List user-visible changes (including defaults/config).  
If none, write `None`.

- Local and Docker onboarding no longer asks for or documents `LLM_PROVIDER`.
- LLM setup is now documented as the OpenAI-compatible trio `LLM_API_KEY`, `LLM_BASE_URL`, and `LLM_MODEL`.
- Existing defaults stay the same: default model remains `gpt-4-turbo-preview`, default base URL remains `https://api.openai.com/v1`.
- LLM integration cases now get 3 total attempts in the runner, so a single semantic mismatch no longer fails the suite immediately.

## Diagram (if applicable)

For UI changes or non-trivial logic flows, include a small ASCII diagram reviewers can scan quickly. Otherwise write `N/A`.

```text
Before:
[user/docker-install] -> [provider + base URL + model + key] -> [provider-specific config branching] -> [ChatOpenAI]
[llm case] -> [single attempt unless transient transport error] -> [assertions]

After:
[user/docker-install] -> [base URL + model + key] -> [direct OpenAI-compatible config] -> [ChatOpenAI]
[llm case] -> [up to 3 total attempts] -> [assertions]
```

## Security Impact (required)

- New permissions/capabilities? (`Yes/No`) No
- Secrets/tokens handling changed? (`Yes/No`) No
- New/changed network calls? (`Yes/No`) No
- Command/tool execution surface changed? (`Yes/No`) No
- Data access scope changed? (`Yes/No`) No
- If any `Yes`, explain risk + mitigation: N/A

### Environment

- OS: Linux (workspace host)
- Runtime/container: local Node.js workspace, backend regression runner, CLI, and targeted `node:test` coverage for LLM retry behavior
- Model/config surface: OpenAI-compatible envs (`LLM_API_KEY`, `LLM_BASE_URL`, `LLM_MODEL`)
- Integration/channel (if any): backend API, CLI bootstrap, Docker env contract, frontend Playwright webServer env, and LLM integration runner
- Relevant config (redacted): `.env` with redacted `LLM_API_KEY`; default `LLM_BASE_URL` and `LLM_MODEL`

### Steps

1. Start from `upstream/master` and create `refactor/remove-llm-provider`.
2. Remove `LLM_PROVIDER` / `--llm-provider` from config, CLI, Docker, workflows, test envs, and docs; add focused backend/env contract coverage.
3. Update the LLM integration runner so every case has 3 total attempts.
4. Run the verification commands listed below.

### Expected

- No source, workflow, test, or documentation path still references `LLM_PROVIDER` or `--llm-provider`.
- Backend config exposes the same defaults without provider branching.
- The LLM integration runner gives each case up to 3 total attempts.

### Actual

- Verified: repository-wide search no longer finds `LLM_PROVIDER` or `--llm-provider`.
- Verified: focused contract coverage, retry unit coverage, backend regression/build/lint, frontend type-check, and CLI help all match the reduced contract and the new retry policy.

## Evidence

Attach at least one:

- [x] Failing test/log before + passing after
- [x] Trace/log snippets
- [ ] Screenshot/recording
- [ ] Perf numbers (if relevant)

## Human Verification (required)

What you personally verified (not just CI), and how:

- Verified scenarios: `node tests/runner.mjs backend-regression`; `npm run build --prefix backend`; `npm run lint --prefix backend`; `npm run type-check --prefix frontend`; `NODE_OPTIONS=--experimental-vm-modules npm exec jest -- --runInBand tests/config.llm-env.test.mjs tests/env-compose.contract.test.mjs`; `node --test tests/llm-integration/lib/*.test.cjs`; `./sclaw help`; `rg -n -S "LLM_PROVIDER|--llm-provider" .`
- Edge cases checked: default LLM model/base URL remained unchanged; Docker/CI/frontend env contracts no longer rely on provider wiring; non-runner callers still keep transient-error-only retry behavior unless they explicitly opt into case-level retries.
- What you did **not** verify: full Docker smoke and real external LLM API calls.

## Review Conversations

- [x] I replied to or resolved every bot review conversation I addressed in this PR.
- [x] I left unresolved only the conversations that still need reviewer or maintainer judgment.

If a bot review conversation is addressed by this PR, resolve that conversation yourself. Do not leave bot review conversation cleanup for maintainers.

## Compatibility / Migration

- Backward compatible? (`Yes/No`) No
- Config/env changes? (`Yes/No`) Yes
- Migration needed? (`Yes/No`) Yes
- If yes, exact upgrade steps:
  1. Remove `LLM_PROVIDER` from local `.env`, CI secrets, and Docker/bootstrap invocations.
  2. Keep or set `LLM_API_KEY`, `LLM_BASE_URL`, and `LLM_MODEL`.
  3. If you used `sclaw docker-install` scripts, stop passing `--llm-provider`.

## Risks and Mitigations

List only real risks for this PR. Add/remove entries as needed. If none, write `None`.

- Risk: downstream scripts outside this repo may still pass `LLM_PROVIDER` or `--llm-provider`.
  - Mitigation: bilingual docs and CLI help now advertise the reduced contract clearly; this PR description calls out the migration steps.
- Risk: retrying case-level assertion failures could delay detection of a persistent semantic regression until the third attempt.
  - Mitigation: the cap is fixed at 3 total attempts, each retry is logged explicitly, and the case still fails if it cannot pass within those 3 runs.

### CODE REVIEW

**Comment by guyi2000** [comment]
✅ **LLM Integration Tests Triggered!**

@guyi2000, I've started the workflow for you.
Please click the link below to monitor progress and **approve the deployment** to the test environment:

🚀 **[View Action Run Details](https://github.com/structureclaw/structureclaw/actions/runs/24277006686)**

File: backend/tests/config.llm-env.test.mjs [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import path from 'node:path';
+import { pathToFileURL } from 'node:url';
+
+import { describe, expect, test } from '@jest/globals';
+
+const repoRoot = path.resolve(process.cwd(), '..');
+const configModuleUrl = pathToFileURL(path.join(repoRoot, 'backend', 'dist', 'config', 'index.js')).href;
```
[the remark this answers is not in the document]
Comment (qinsz01) [reply backend/tests/config.llm-env.test.mjs]: Updated `backend/tests/config.llm-env.test.mjs` to resolve `../dist/config/index.js` from `import.meta.url` instead of `process.cwd()`, so the test stays stable regardless of the working directory. Re-ran `NODE_OPTIONS=--experimental-vm-modules npm exec jest -- --runInBand tests/config.llm-env.test.mjs tests/env-compose.contract.test.mjs` in `backend/` after the change.

**Comment by guyi2000** [comment]
✅ **LLM Integration Tests PASSED**

@guyi2000, the test run has completed.

```
Results: 42/42 passed, 0 failed
Total time: 1097.9s
============================================================
```

📦 **[Download Logs](https://github.com/structureclaw/structureclaw/actions/runs/24277006686)** or check the Artifacts section.

File: backend/tests/env-compose.contract.test.mjs [hunk 2]
```diff
@@ -72,7 +72,6 @@ describe('env example vs docker-compose contract', () => {
     expect(dockerInstall).toBeDefined();
     expect(dockerInstall.usage).toContain('sclaw docker-install');
     expect(dockerInstall.usage).toContain('--non-interactive');
-    expect(dockerInstall.usage).toContain('--llm-provider <name>');
     expect(dockerInstall.usage).toContain('--llm-base-url <url>');
     expect(dockerInstall.usage).toContain('--llm-api-key <key>');
     expect(dockerInstall.usage).toContain('--llm-model <name>');
```
[the remark this answers is not in the document]
Comment (qinsz01) [reply backend/tests/env-compose.contract.test.mjs:78]: Added the negative assertion in `backend/tests/env-compose.contract.test.mjs` so the contract now locks both the remaining non-interactive flags and the absence of `--llm-provider`. Re-ran `NODE_OPTIONS=--experimental-vm-modules npm exec jest -- --runInBand tests/env-compose.contract.test.mjs` in `backend/` after the change.

**Review by guyi2000 — APPROVED** [review]
